### DISCUSSION TITLE
请教：下拉选择框使用JQuery动态添加条目时无效

### DISCUSSION THREAD

**Discussion by HuangLei0x2e5** [layui/layui#1684] [讨论]
我遇到一个问题：我下拉框里的数据要从后台获取，但是在ajax成功回调函数里使用JQuery 动态添加option无效，但是在页面上直接添加有效

我的代码：

<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Test </title>
    <meta name="renderer" content="webkit" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link
      href="//cdn.staticfile.org/layui/2.9.6/css/layui.css"
      rel="stylesheet"
    />
  </head>
  <body>
    <form class="layui-form layui-row layui-col-space16">
      <div class="layui-col-md6">
        <select
          id="appoint-select-add"
          lay-search=""
          lay-filter="appoint-select-filter"
        >
          <option value="1">AAA</option>
        </select>
      </div>
    </form>

    <script src="layui/layui.js"></script>
    <script>
      function loadAPPoints() {

        // 如果放在这里能正常执行
        // var select = layui.$("#appoint-select-add");
        // select.append("<option value='abc'>CCCCC</option>");

        layui.$.ajax({
          url: "debug/appoint.json",
          type: type,
          dataType: "json",
          data: data,
          success: function (response) {
            // 放在这里无效
            var select = layui.$("#appoint-select-add");
            select.append("<option value='abc'>CCCCC</option>");
          },
          error: function (jqXHR, textStatus, errorThrown) {
            layer.msg(errorThrown);
          },
        });
      }
      loadAPPoints();
    </script>
  </body>
</html>

**Comment by bxjt123** [comment]
动态添加后，需要执行`form.render('select')`重新渲染下，参考[文档](https://layui.dev/docs/2/form/#render)

**Reply by HuangLei0x2e5** [reply]
按您的方式改了，现在可以了，感谢

**Comment by LSL1618** [comment]
layui的select组件不是数据驱动视图模式，选项数据变更之后需要render()重新渲染。

**Reply by HuangLei0x2e5** [reply]
我是做后端的，现在兼职做前端，对前端的技术栈和特性不熟悉，边学边做。
感谢
